add blog post for release of hpy-0.0.4

fangerer opened the pull request:
In preparation to the upcoming release, I've again prepared a blog post.

cfbolz commented:
sounds great to me. maybe an extra advantage in the bullet list at the top could be "evolvability" or something along these lines. the fact, that it decouples the CPython internals from the exposed API, and the fact that on the other hand hpy will be a much more stable api.

hodgestar commented on posts/2021/10/hpy-0.0.4.md:
```diff
@@ -0,0 +1,219 @@
+support, better support for native fields (HPyField) and global variables
+(HPyGlobal), 
+
+Great new also is that we are now able to provide two more non-trivial projects
+that have been (partially) migrated to HPy. This is,
+[Kiwisolver](https://github.com/hpyproject/kiwi-hpy/)
+and [Matplotlib](https://github.com/hpyproject/matplotlib-hpy/).
```
```suggestion
Great news too is that we are now able to provide two more non-trivial projects
that have been (partially) migrated to HPy. This is,
[Kiwisolver](https://github.com/hpyproject/kiwi-hpy/)
and [Matplotlib](https://github.com/hpyproject/matplotlib-hpy/).
```

timfel replied:
I would also add `We are still cleaning these ports up and will write another blog post about the ports and open them for discussion with the project owners.`

hodgestar commented on posts/2021/10/hpy-0.0.4.md:
```diff
@@ -0,0 +1,219 @@
+the Python implementation may use an arbitrary internal representation. This
+means that the UTF8 representation is just temporarily created for this API call
+and so the raw data must be released at some point. The contract here is that
+the raw data pointer is as long valid as the corresponding handle is valid.
+Example:
```
```suggestion
the raw data pointer is valid as long as the corresponding handle is valid.

Example:
```

hodgestar commented on posts/2021/10/hpy-0.0.4.md:
````diff
@@ -0,0 +1,219 @@
+```
+
+
+It
+is easy to forget about this resriction and if the raw data pointer is used
````
```suggestion
It is easy to forget about this resriction and if the raw data pointer is used
```

hodgestar commented on posts/2021/10/hpy-0.0.4.md:
```diff
@@ -0,0 +1,219 @@
+HPy 0.0.4 is out! The third official HPy release comes with many new features
+and was again made available on PyPI (first time with binary wheels).
+
+Besides new API functions, the major highlights of this release are Python 3.10
+support, better support for native fields (HPyField) and global variables
+(HPyGlobal), 
```
I would add a few more things to this list -- maybe mention one or two of the new API functions as examples and that there are new features in debug mode. Otherwise the introduction makes it sound like not too much happened when a lot did happen.

hodgestar reviewed (approved):
This looks great! I left some small suggestions for improvements.

fangerer commented:
Thanks for your comments and sorry for the delay. Since we are now up to publish the release on PyPI, I want to finish this blog post.
@hodgestar I've applied your suggestions. Please have another look at the teaser.
@cfbolz I think the *evolvability* does not belong to the release blog post. So, I've added a bullet point to the main page. Please see `index.rst` for that. I wasn't sure about how much we want to go into details there.

hodgestar reviewed (approved):
Looks good to me, but the CI robots seem unhappy. Happy for it to be merged once the robots are satisfied.

steve-s commented on posts/2022/05/hpy-0.0.4.md:
````diff
@@ -0,0 +1,240 @@
+```
+
+
+Iu is easy to forget about this resriction and if the raw data pointer is used
````
Typo? "It is easy..."

steve-s commented on posts/2022/05/hpy-0.0.4.md:
```diff
@@ -0,0 +1,240 @@
+pointer will then cause a crash of the application. Unfortunately, not all
+systems support that feature (since that needs OS support), we use a different
+strategy on other systems and fill the pointer with some marker bytes that make
+it easy to detect.
```
To be fair, I think Windows can support this. We just did not implement it yet.

steve-s commented on posts/2022/05/hpy-0.0.4.md:
```diff
@@ -0,0 +1,240 @@
+
+HPy functions that are called from Python receive handles that are owned by the
+caller. This means that those handles must not be closed by the callee but it
+is, of course, possible to call `HPy_Close` on them. For example:
```
"..to erroneously call..." to make it clear that's the bug we're talking about?

steve-s commented on posts/2022/05/hpy-0.0.4.md:
```diff
@@ -0,0 +1,240 @@
+HPyDef_METH(foo, "foo", foo_impl, HPyFunc_O, .doc="closing argument")
+static HPy foo_impl(HPyContext *ctx, HPy self, HPy arg)
+{
+    // 'arg' is owned by the caller
```
// error: 'arg' is owned by the caller

fangerer commented:
@hodgestar I've addressed all comments but as you already pointed out, _Nikola_ is not happy. It says:
```
ModuleNotFoundError: No module named 'doit.cmd_auto'
```
Can you help me a bit?

fangerer commented:
Okay, thanks, I'll try to fix that.
